projects/packages/my-jetpack/_inc/components/welcome-flow/ConnectionStep.tsx

```diff
@@ -16,23 +21,50 @@ type ConnectionStepProps = {
  *
  * @param {object} props - ConnectioStepProps
  * @param {Function} props.onActivateSite - Alias for handleRegisterSite
+ * @param {Function} props.onUpdateWelcomeFlowExperiment - Updating the welcomeFlowExperiment state
  * @param {boolean} props.isActivating - Alias for siteIsRegistering
  * @returns {object} The ConnectionStep component.
  */
-const ConnectionStep = ( { onActivateSite, isActivating }: ConnectionStepProps ) => {
+const ConnectionStep = ( {
+	onActivateSite,
+	onUpdateWelcomeFlowExperiment,
+	isActivating,
+}: ConnectionStepProps ) => {
 	const { recordEvent } = useAnalytics();
 	const { setNotice, resetNotice } = useContext( NoticeContext );
 
 	const activationButtonLabel = __( 'Activate Jetpack in one click', 'jetpack-my-jetpack' );
 
-	const onConnectSiteClick = useCallback( () => {
+	const onConnectSiteClick = useCallback( async () => {
 		recordEvent( 'jetpack_myjetpack_welcome_banner_connect_site_click' );
-		onActivateSite().then( () => {
-			recordEvent( 'jetpack_myjetpack_welcome_banner_connect_site_success' );
-			resetNotice();
-			setNotice( NOTICE_SITE_CONNECTED, resetNotice );
-		} );
-	}, [ onActivateSite, recordEvent, resetNotice, setNotice ] );
+		onUpdateWelcomeFlowExperiment( state => ( { ...state, isLoading: true } ) );
+		await onActivateSite();
+
+		recordEvent( 'jetpack_myjetpack_welcome_banner_connect_site_success' );
+
+		await sideloadTracks();
```

Comment:
We could run some code version flag? However, I think sideloading Tracks will be useful in general now once we've introduced async connection - otherwise we lose all Tracks events that happen in the first user session (following site connection).

Reply:
True - let's leave it as is in this PR

Reply:
> However, I think sideloading Tracks will be useful in general now once we've introduced async connection - otherwise we lose all Tracks events that happen in the first user session (following site connection).

This is correct. I think we can improve how we sideload tracks after async connections, but for now the call is necessary even when not in an experiment.